### PR TITLE
Add react ref and key props to components

### PR DESCRIPTION
**Pull request by Freddy03h**
- Add ref prop to NavigationContainer (important for universal link scenario)
- Add key prop to components

### CODE REVIEW

File: src/Native.re [hunk 1]
```diff
@@ -99,6 +99,7 @@ module NavigationContainer = {
   [@bs.module "@react-navigation/native"] [@react.component]
   external make:
     (
+      ~ref: ReactNative.NativeElement.ref=?,
```
Comment (MoOx) [inline src/Native.re:102]: This one seems unecessary

**Review by MoOx — CHANGES_REQUESTED** [review]
That's weird. We should not have to define key, they should be automatic with the jsx ppx...
But I made a test & the compiler is yelling indeed.
We should only add the ref when we create the makeProps by hand and not when we have `[@react.component]` (= ppx, so auto ref)

**Comment by Freddy03h** [comment]
The doc say the ppx add the `key` but not the `ref` : https://reasonml.github.io/reason-react/docs/en/components#reactcomponent

File: src/Native.re (hunk 1 above)
Comment (Freddy03h) [reply src/Native.re:102]: It doesn't work without it, and I looked on other project to do it : https://github.com/reason-react-native/datetimepicker/blob/f479766c79ce24248c8752c2b1a6e5edb35bb2b1/src/ReactNativeDateTimePicker.re#L21

**Comment by Freddy03h** [comment]
I added `key` only for components made with `makeProps`

**Comment by MoOx** [comment]
Oh that's right, I read too quickly!

**Comment by MoOx** [comment]
5.0.1 is out :)

**Pull request merged**